src/frontend/src/optimizer/plan_node/mod.rs

```diff
@@ -318,13 +331,25 @@ impl dyn PlanNode {
     }
 
     /// Explain the plan node and return a string.
-    pub fn explain_to_string(&self) -> Result<String> {
+    fn explain_to_string(&self) -> Result<String> {
+        // In order to let expression display id started from 1 for explaining.
+        // We will reset expression display id to 0 and clone the whole plan to reset the schema.
+        let plan = {
+            let old_expr_display_id = self.ctx().get_expr_display_id();
+            self.ctx().set_expr_display_id(0);
+            let plan = PlanCloner::clone_whole_plan(self.clone());
+            self.ctx().set_expr_display_id(old_expr_display_id);
```

Comment:
Agree with @st1page. A little counterintuitive to me and I am not sure if it is worth it.

Reply:
This `clone_whole_plan`  is restricted in `Explain` scope only, so I think it is acceptability. Otherwise it is painful to add a new optimization, because it would affect lots of plan tests.